tests/test_time_series/test_forecasting/test_window_features.py

```diff
@@ -0,0 +1,227 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from feature_engine.timeseries.forecasting import WindowFeatures
+
+
+@pytest.mark.parametrize("_windows", [3, "1H", "15min", 5])
+def test_permitted_param_windows(_windows):
+    transformer = WindowFeatures(window=_windows)
+    assert transformer.window == _windows
+
+
+@pytest.mark.parametrize("_windows", [True, [4, "hola"], 4.3])
+def test_error_when_non_permitted_param_windows(_windows):
+    with pytest.raises(ValueError):
+        WindowFeatures(window=_windows)
+
+
+@pytest.mark.parametrize("_functions", [np.mean, np.std, np.max, np.median])
+def test_permitted_param_functions(_functions):
+    transformer = WindowFeatures(function=_functions)
+    assert transformer.function == _functions
+
+
+@pytest.mark.parametrize("_functions", ["che", True, np.sin, 1984])
+def test_error_when_non_permitted_param_functions(_functions):
+    with pytest.raises(ValueError):
+        WindowFeatures(function=_functions)
+
+
+@pytest.mark.parametrize("_periods", [1, 2, 3])
+def test_permitted_param_periods(_periods):
+    transformer = WindowFeatures(periods=_periods)
+    assert transformer.periods == _periods
+
+
+@pytest.mark.parametrize("_periods", ["pizza", 3.33, True, None])
+def test_error_when_non_permitted_param_periods(_periods):
+    with pytest.raises(ValueError):
+        WindowFeatures(periods=_periods)
+
+
+@pytest.mark.parametrize("_freqs", ["1h", "90min"])
+def test_permitted_param_freq(_freqs):
+    transformer = WindowFeatures(freq=_freqs)
+    assert transformer.freq == _freqs
+
+
+@pytest.mark.parametrize("_freqs", [5, "asado", [1, 2, 4], False])
+def test_error_when_non_permitted_param_freq(_freqs):
+    with pytest.raises(ValueError):
+        WindowFeatures(periods=None, freq=_freqs)
+
+
+def test_get_feature_names_out(df_time):
```

Comment:
is this test identical to the one for lag features? if yes, let's move it to the common tests file, and pass the transformer as an argument with parametrize.

The common test file is where we check the tests from check_estimator.

Reply:
It's different because "window_X" is included in the transformed variables. See the comment above.